### ISSUE TITLE
EventAdapter.beforeChangeValueOf issue

### ISSUE THREAD

**Issue by mwiesiolek** [FluentLenium/FluentLenium#503]
Hi,

I am trying to execute one of my end to end tests and I receive stacktrace as following:

```
java.lang.AbstractMethodError: org.fluentlenium.core.events.EventAdapter.beforeChangeValueOf(Lorg/openqa/selenium/WebElement;Lorg/openqa/selenium/WebDriver;)V

	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.openqa.selenium.support.events.EventFiringWebDriver$1.invoke(EventFiringWebDriver.java:80)
	at com.sun.proxy.$Proxy134.beforeChangeValueOf(Unknown Source)
	at org.openqa.selenium.support.events.EventFiringWebDriver$EventFiringWebElement.clear(EventFiringWebDriver.java:363)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.fluentlenium.core.proxy.AbstractLocatorHandler.invoke(AbstractLocatorHandler.java:294)
	at org.fluentlenium.core.proxy.AbstractLocatorHandler.invokeWithRetry(AbstractLocatorHandler.java:280)
	at org.fluentlenium.core.proxy.AbstractLocatorHandler.invoke(AbstractLocatorHandler.java:272)
	at org.fluentlenium.core.proxy.ComponentHandler.invoke(ComponentHandler.java:98)
	at com.sun.proxy.$Proxy139.clear(Unknown Source)
	at org.fluentlenium.core.domain.FluentWebElement.clear(FluentWebElement.java:225)
	at org.fluentlenium.core.domain.FluentWebElement.write(FluentWebElement.java:247)
	at org.fluentlenium.core.domain.FluentListImpl.write(FluentListImpl.java:306)
	at org.fluentlenium.core.action.Fill.with(Fill.java:38)
	at pl.mw.core.controller.SettingControllerE2ETest.beforeMethod(SettingControllerE2ETest.java:51)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:75)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:86)
	at org.fluentlenium.adapter.junit.FluentTestRule$1.evaluate(FluentTestRule.java:44)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:84)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:252)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:94)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70)
	at org.fluentlenium.adapter.junit.FluentTest$2$1.evaluate(FluentTest.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:191)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:51)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:237)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
```

My test code looks as following:

```
    @Before
    public void beforeMethod() {
        window().maximize();

        goTo(getUrl("setting/home"));

        el(USER_NAME_FIELD_SELECTOR).now()
                .fill()
                .with(env.getProperty("user.1.username"));

        el(PASSWORD_FIELD_SELECTOR).now()
                .fill()
                .with(env.getProperty("user.1.password"));

        // logs in
        el(LOG_IN_OUT_BUTTON_SELECTOR).now()
                .click();
    }

    @After
    public void afterMethod() {

        // logs out
        el(LOG_IN_OUT_BUTTON_SELECTOR).now()
                .click();
    }

    @Override
    public String getWebDriver() {
        return "phantomjs";
    }

    @Test
    public void shouldSetNewWebShopName() throws Exception {

        // given
        String webShopName = String.format("web-shop-name-%s", System.currentTimeMillis());
        String inputFieldSelector = "input[data-id=\"web-shop-name\"]";
        String submitButtonSelector = "form[action=\"/setting/webshopname\"] button[type=\"submit\"]";
        String webShopNameSelector = ".navbar-brand";

        // when
        el(inputFieldSelector)
                .fill()
                .with(webShopName);

        el(submitButtonSelector).click();

        // then
        assertThat($(webShopNameSelector).first().text()).isEqualTo(webShopName);
    }
```

When 

```
el(USER_NAME_FIELD_SELECTOR).now()
                .fill()
                .with
```

is executed it fails.  I found some similar issue here: https://github.com/FluentLenium/FluentLenium/issues/457

but I don't see that transitive dependency is a problem in my case. Since, I am able to execute my other end to end tests.

Below you may find part of my gradle configuration:

```
    testCompile('org.springframework.boot:spring-boot-starter-test')
    testCompile('org.fluentlenium:fluentlenium-junit:3.2.0')
    testCompile('org.fluentlenium:fluentlenium-assertj:3.2.0')
    testCompile('junit:junit:4.12')
    testCompile('org.seleniumhq.selenium:htmlunit-driver:2.25')
    testCompile('xml-apis:xml-apis:1.4.01')
    testCompile('com.codeborne:phantomjsdriver:1.2.1')
```

Whereas, spring-boot-starter-test has 1.5.3.RELEASE version.

Thanks in advance for your help.

**Comment by mwiesiolek** [comment]
I've investigated the issue more and maged to fix it.

My current gradle configuration looks as following:

```
    testCompile('org.fluentlenium:fluentlenium-junit:3.2.0')
    testCompile('org.fluentlenium:fluentlenium-assertj:3.2.0')
    testCompile('org.seleniumhq.selenium:htmlunit-driver:2.21')
    testCompile('xml-apis:xml-apis:1.4.01')
```

What I've done was:

1.  Removed     testCompile('com.codeborne:phantomjsdriver:1.2.1') dependency
2.  Downgraded org.seleniumhq.selenium:htmlunit-driver:2.25 dependency to 2.21 (while clearing input field in version 2.25 throws:  com.gargoylesoftware.htmlunit.html.HtmlInput.setValueAttribute  NoSuchMethodException)

Work with spring boot 1.5.3.RELEASE
